src/components/Context/injectEnvironment.js

```diff
@@ -14,14 +14,24 @@ import { deepAssign } from "../../utils";
 
 export default window => {
   return xdm => {
-    const { innerWidth, innerHeight } = window;
+    const {
+      document: { documentElement: { clientWidth, clientHeight } = {} }
+    } = window;
+
     const environment = {
-      type: "browser",
-      browserDetails: {
-        viewportWidth: innerWidth,
-        viewportHeight: innerHeight
-      }
+      type: "browser"
     };
+    const hasWidth = clientWidth && clientWidth >= 0;
```

Comment:
This probably isn't what you're intending. If `clientWidth` is `0`, then `hasWidth` will end up being `0`, which means `environment.browserDetails.viewportWidth` won't be set. It's probably worth adding a unit test for this case.

Reply:
Thanks for finding this!